### ISSUE TITLE
Compression corrupt during view staleness calculation

### ISSUE THREAD

**Issue by Bolebo** [gesellix/couchdb-prometheus-exporter#238]
Hello,

I have a huge collection with a huge update_seq.
I recently updated CouchDB from 3.2.0 to 3.2.3 and the exporter now fails with this error : 

Vm-TIS-Pxp-couchdb1 couchdb-prometheus-exporter[3260]: E0607 09:58:53.340735    3260 collector.go:425] Error collecting stats: compression corrupt

After some digging, I found the error occurs when decoding the update_seq of the collection in the erlang.BinaryToTerm(payload[:n]) conversion. This decoding is used only in the View Staleness counter calculation.

I don't use this particular metric, but I definitely use most of the other metrics. But because of this error, the exporter is stucked.

Do you have any solution ?

**Comment by gesellix** [comment]
Hey,

could you try to get an example of a failing update_seq so that I can check it locally or as an explicit test case?
I currently don't know why or how the decoding would fail.

Thanks!

**Comment by Bolebo** [comment]
Sure, here it comes : 
8181733470-g1AAAMsDeJzV3QfQHkUZAOAfEkQUDUVFVECQDgm3t3tbRGmCgihdQAVxKyE9IcEASQg1ShERRARUUJFgwYKIiCAWFEQEFTtIiRRBRJSuJn77v5vvm7vM3M14-e7fhZkwk5nMn4fd23vf3XffmzYyssbI5HFmZH89c56ebNRuh06feMg-B088YP6sieG30CQ5b8Y8PDfDdspce5zO5jh8NJqDde7MJDl_0jEz5to5M-S0SXrazHlGzpo1aYadO21kZGTNGeN7v46s6P1nRe-fKZPHqZEJszef3vu9tWlBlMV09f_gIWOWA2Z5CZMRajPLO8eQlpj_Aua_o5hxi88YxSBeZNyo1f-DGzBFS8x_APOfgJkB0wwVkjq9-n_wkEfm34D5N2BOOW8UYzQjDNVjhjEybZ-ZFwHzIjwzx94-iiFS5DI3yT0zLwDmhTDNFo9icmEoMbLzkclaYp4HzPMeMz6_51J4ZhzLVcFW_w8e8sg8B5jnYGROWmsUY3MqMpQnt5o9C5hnwwKwZBTDLSuEsZ2vZm2n2TOAeWZ0muFb8lGMQxxxUY8ZxjRri3kaME-XMBobgWX9ajYMTNul-V-A-RcszVPnjGKEk05blFxs9k_A_LOEob2lWZL0ME8B5qkShhCrnEkP8w_A_KOEQU5gLtOLmp8EzJNhaT4KUoDMcCbqF4BhvGfylpi_A-bvMDJzToX3DENSk_qX5v_1g4f80nwCME_AyJw2G54ZbYxy9ZlmjCnA3wDzN8As2h4CTW0sVbjzadYW8zhgHg9R83KYZjhXhnePabuaPQaYx8JqdjZMM0FJ1hDOxLia_RUwfw3T7HJYzZCk3NQvzTGOzKOAebQ0MlYXLMtJciPzCGAeCSOzE4wMc7in6XxkcEvMw4B5OCwAN4ZwRhHXe3Ou9h885BTgIcA8NJoCFBPGAyYrMEbp5TN_AcxfShhUZILi-mkW4x7AMsAsC8_MIbBv5jR3OL3Y7EHAPBhWswdDpllYwesxw4jN2o7MA4B5IOzOQKApi17W7Lrfa277zNwPmPvDM6NGMUUmM666f2baYu4DzH2AWWcXeGaYdY53v5q1jQD-DJg_h_OZRSE5yynFIrm0-V7A3FvC9B5vKmzROabtNLsHMPeUphm3hZC6PgIYxjRrm2n-CTB_CnsA2wAGSUtRfdQcY6b5R8D8ETAnXwkRgClMnnd_2NR2ZP4AmD-EQBM2zm1WGJLX75sNY2TaRs2_B8zvAbNwAWyca2IVr89nhhE1t51mvwPM78LIPAKrWZZb4-qX5hjzmd8C5rch0IR8RiOj_S5gaiPzG8D8JiwAsG9WOEozWv_MxLgA3A2Yu0OguR8kZ723prH175lhLABtI4BfA-bXEAHMeBRGRmXUue43NNo-M78CzK_CyNwE7xlpmcnrX5ox7gH8EjC_HA1nyJ0PAQYbjUh9HUCMKcBdgLmrhLFFQWzDyVmMmDsBcydgVkyA2MwwlJv6veYYMb8AzC9KGOuI7AUAyWHuAMwd5ZEpJFeqPtCMEfNzwPy8hGFMG0LqV7MY85nbAXN7iM02DodN2kpTH87EmM_8DDA_C--Zm2EPwOVYNsRmMW413QaY28I0WxciAG4l1_V7ADE-M7cC5tYSBhWGGt39S7PtNPspYH4akrPHYEeTOexE92eabReAnwDmJ-GZeWYUozTBmNZPsxjT5lsAc0vAbBXCGYYy6ZJLzn4MmB8HDAnl84hS1H1y1jaf-RFgfgT5zKwnITlThhmU3vbsDwHzwxLGCsklrV8AYsT8ADA_KGFclttc1qfNw8C0Xc1uBszNYXfmilAJKDFl3VfPth2Z7wPm-zAy0-GwCWWZ4qr74rm2Z5o3AeamEsa5AquGKyfDONNsG87cCJgbIZxZPjcsALnluvsUoC3me4D5XglT8N7_-YalOUbMDYC5ATDPbgnTTFhnGo4BY8R8FzDfLWEypYU16e0BXA-Y60sY5IyhtL5GM0bMdwDznRJGak5yXr-axYi5DjDXAeZWHe7PMCESXM2-DZhvlzDUKSJFeiNzLWCuLWGU1rmR6e0BfAsw3yphMmdpXnQ_Mm2Ts2sAc004OYMKDZ5jR0n9BboYi7S_CZhvhk3A6-AyUC9sFiy9C3TfAMw3wu7Mi7CjiXvBTEP5_DBSgLZR89cB83WImmfuG_IZbDLd_S2NtvnM1wDzNcBMvgoWAC0YzbqvamqLuRowV5cwhltKsvRuNn0VMF8NW03nQ2ymJM5s9wtA22fmK4D5SnlkJJUE1y8AMVbPfhkwXw5L810QAbiCKpte7cyXAPOlsDQXsABwwzPd_fZs23DmKsBcBeHMZQymWSE40ellmksBs7SEcRwbg9OrA7gSMFdCwek2e4dODZajhiONGDFfBMwXS5hebCYsTe_KyRWAuaKE4TkrDO--7URbzBcA84USRnHjjEgP83nAfB4wO54WdjSdonl6q9nnAPO5gDkTFgCrmUTpFQJdDpjLAyb0nSmoFDy9erPLAHNZCSMNz7MxeGm2jc0-C5jPhthsj7BvplhTQ5AYo-bPAOYzEDU7uNigCXfG1k-zGPsBfBown4aRWXBYuHKCncnql-YYq5ouBcylIWrGoRRYEJPVb5zHWAdwCWAugWl29EwINBmXeAy6m7SdZhcD5uKQNkORtmA5sia9ivNPAeZTpZFhmrqCp1ehcRFgLiphKMukk-lhPgmYTwJmymUQATDLdV6_cR4j5kLAXFjCcFGwwnbfE7BtOPMJwHwCwpktLoSXZo5yNAYtwdpiLgDMBSUMdZbpPL0D2vMBcz5gtoYmOlYylpn0Lmp_HDAfL2FyXlhl6vfNYsScB5jzShite4uzSu-A9mOA-Vh5muHCyIZAM0bMuYA5FxocPnDJKAYLmamGY8AYMR8FzEfLGNqbZw3VszFizgHMOSVMLmmGG4rnYsScDZizSxikJM_G4GJD2xTgLMCcFVKA0KoFUU4bes8OIwVoizkTMGeGtPm5cLURZUakdxfgI4D5SBiZq2FpJjSjDWeawzhsaov5MGA-HDAbwF6z6KXNpP4C3TAwbVOAJYBZEooaboBAkxZWNDQEifFu8xmAOSNsAl4cdmfyHKP6FCDGKyenA-b0MM2g8xxD2IgE-wGcBpjTSkUN1BGBGi4Dxbg7cypgTg39AC4O27NK46x-AYjxLsApgDmljMG4MdCM8S7AyYA5ObQF3zMszVrQBGOzxYBZXMJYonPFUjpsUif1flUnrYBvz4wv9lo_9DbLHecp5TJqkYcsqkK4KAp_pyEhyEIPWViFOKl4LlI6YVYLPGTBKhBFCFHd9zNrATnRQ07sQw66J9zIkDpvCPcjg5zgISdUITZXGW04VIoMcryHHF-FZFqZzCW1as33kPlViMwEEg0tjOM651cf8pAPBcjIuEXnhpbf2ClW_4zEVeevjvOQ4_qQk-ZDniJzgVD9eySuQlI1z0Pm9SGnLgNITrWj9fFWXHm9mushc_uQxSfC6XFmCMlSyhzVsR5ybB-ycCT0XpaY0JSuj6g5HjJnAFkejiYyhVVKZSNqtofMHqxaBoJGzZXW9d3j4qpMUrM8ZNZKyIRjZsEL0TrKG766GFctv5rpITMHIwIliSgnymbd1_G3GJEZHjKjPyKTScgQs94f7b43eYsQxf-91fSVIQpZsSx8mMAx03CBJ7JYy_-qplUh2gink7q_r6Z6yNQ-5GnYFpLIZVokFf1O8ZApVQgS0sqkusSrYzzkmCpE0N5wFCn17VOTPWRyFZIx4XKVVIZ4tIccvRJCj7sbYi2TSaRSKtJRzkNcFaIwFjlOavm1HmKrEG0yrFlKNWC93KkHMVVITlHOx2CDrkUYrz1ED2KtE2BEmBSW1WeIcZ3JK-Uhqg85-RRYfglXWKZUXKCkh8hBqhs6jAjhXEOqG1li9UEP-eBg8-HwEMY7ljW8RyJLdY_ykKMGqe7UcBVXU9fw8afIptYHPOQDg2fk9NC-DmWk4UvWkU2tIz3kyMGqdT8UrCkrdcO3xeK65qGO8JAj-hnidGgpzCRBnNeHKHEdVav3e8j7-5ApAgoIsGASd9-yrsWIvM9D3leFSG1ULyVJqOZevddD3rsyREEvLIRnhCiHeVJh_OEecngVInsrlk3rDPEwDzlsJSTfZGmAaJmJpDLEQz3k0CqEGEtzm1Ilt3qPh7ynCimURK7hHldkkEM85JBVphbDhDV02Y0McrCHHNyHbPg4vEdyhPKGzwZHdhh6kIcc1A9RFmwWPrCRy4zVP-xx9ddXB3rIgdURyZSmliZV5nSAhxxQhVjNmR6D-_Qtwvj9PWT_wdS6F1YtnKFCdP_1thbPyH4est8gjIdbdMwXPpikjt7e7SHvHhy9wTUNmWspXFJHb-_ykHcNRmRHgDhcSNd9q8YWz8i-HrJvFWKk0lx2n-q2qHx4p4e8c7AddBu8ELXEVHdfwtFiau3jIfsMVq0jYKfRZobr7juAtMgQ9_aQvQcP-8rvgGmCi-6_aNBiRN7hIe8YPCO7h95fwjrU_fLbYkTe7iFvH-TsF4RWWUgI1P0lmRYhyl4eslc_RNnhvnCrlDmbVEd2taeH7FmFWJZnNKk22eptHvK2PmT7pSGMxxKrpEZkDw_Zowox1umCJXUYuruH7F6FOJlZZpKqxN7NQ3arQrSjzmVJ5ey7esiufciBAFGKK1skVcKxi4fsUoX4L6-bIqmdxrd6yFurEIGl0yKp5fctHvKWKoT0Xoaq4Xg6spx9Zw_ZeRDGw0GPkkIUuvtO6y2Cxjd7yJsHkE1XVgchw-ufkcgyROEhYgDZFnZRCoTQGBQMtBgR7iF8cFthLzhnpzxDDRV0kY0I8xA2gFwLzwhT0uruP0jWAkI9hA4qH5bAe8QqaRpOdSObWoWHFH3IGeuFnUZeCJJSO0VFPIT0M8TjNwq7KM42bdBFds6OPQT3IdN2halFen-2oV93ZOfsuYfkVQiljBtRv_kQWc6OPAT1Q5Rdr4ecnWn_ye6UYq3MQ7IqhGeENLUaiQyyk4fs1Idsd39ommQpkkmlupM8ZFIVYhzz301ICTLRQyZWIRYpQ2RSqe6OHrJjFZKpnEua1MO-g4fsMDgMvTW0SaRENbTjjQyyvYdsX4VIiXPDk-r5sJ2HbFeFENf7N61alG09ZNtVRoQJZYvuC89ahPHbeMg2g_OREGtJhCjpvsVbi6Bxaw_ZulplyjWSuOj-HmKLw9CtPGSrQYa4Ava1dIF7K3Dnh6EtIFt6yJaDzQcoKTcYE9rQOyiuHnXqTR7ypv7UmjobXohGI4q677fdYkS28JAtBg_7HZCPEFkYUh-iRDYim3vI5tURMVrofAy-GtAC8kYPeWMVQnNVoIbD0Mggm3nIZoPj6QOgzIlYWzR8zDUyyKYesmkV4pTKFU-pH73axEM26ZeUL4FWCZrrXI7BsUILyBs85A0DyMxwWYwqIZMqzny9h7y-D1kKjcIUZUSl1brtdR7yuipEs1wUDbFWZJCNPWTjKkQ6i61LakRe6yGvrUJwLjR2Se2ibOQhG_UhT90QRoQbIpOCvMZDXlOFMOOKpstikUFe7SGvrkI0QblO6isA6lUe8qoqBBFHCtb9dlCLMH5DD9lwcMX1IggaJSNZw7cZIyuX3cBDNhikusfCm52jgo3B_ZEWI7K-h6w_SHWvgYc9d1Q0XN-LLGdfz0PWG0ytm2D5pa4gDd2cIssQJ3jIhH4YP-sW2PslijWVOUWWj7zSQ15ZhaBC4byhgi6yzYdXeMgrBg_7kZBYGVOIhgYvw5haLXYa1_WQdVcZkYxbUtQHjZEdT7_cQ14-gDwMQSO1mebdXwNvAXmZh7xsAHki9EXJERuDZ6RFiLKOh6zTD1GWTYKdRiE1S-vo7aUe8tIqhBPJlEvqoGdtD1m7D3l853CfnWiRVuXDSzzkJVUI4owhmlTBwFoeslYV4gpBecMGXWQQ_30FNb4KMZQK1tByJzLIOA8Z14c8zyAf6T3tsuHjqpFB1vSQNauQjGNO0mpvuIaHrLHKiEha8CKlVUuOqJGqwtisQGQIfRKm_A-WO5nR

**Comment by gesellix** [comment]
Note to self to reproduce via cli:

```
 go run cli/update_seq/decode.go 8181733470-g1AAAMsDeJzV3QfQHkUZAOAfEkQUDUVFVECQDgm3t3tbRGmCgihdQAVxKyE9IcEASQg1ShERRARUUJFgwYKIiCAWFEQEFTtIiRRBRJSuJn77v5vvm7vM3M14-e7fhZkwk5nMn4fd23vf3XffmzYyssbI5HFmZH89c56ebNRuh06feMg-B088YP6sieG30CQ5b8Y8PDfDdspce5zO5jh8NJqDde7MJDl_0jEz5to5M-S0SXrazHlGzpo1aYadO21kZGTNGeN7v46s6P1nRe-fKZPHqZEJszef3vu9tWlBlMV09f_gIWOWA2Z5CZMRajPLO8eQlpj_Aua_o5hxi88YxSBeZNyo1f-DGzBFS8x_APOfgJkB0wwVkjq9-n_wkEfm34D5N2BOOW8UYzQjDNVjhjEybZ-ZFwHzIjwzx94-iiFS5DI3yT0zLwDmhTDNFo9icmEoMbLzkclaYp4HzPMeMz6_51J4ZhzLVcFW_w8e8sg8B5jnYGROWmsUY3MqMpQnt5o9C5hnwwKwZBTDLSuEsZ2vZm2n2TOAeWZ0muFb8lGMQxxxUY8ZxjRri3kaME-XMBobgWX9ajYMTNul-V-A-RcszVPnjGKEk05blFxs9k_A_LOEob2lWZL0ME8B5qkShhCrnEkP8w_A_KOEQU5gLtOLmp8EzJNhaT4KUoDMcCbqF4BhvGfylpi_A-bvMDJzToX3DENSk_qX5v_1g4f80nwCME_AyJw2G54ZbYxy9ZlmjCnA3wDzN8As2h4CTW0sVbjzadYW8zhgHg9R83KYZjhXhnePabuaPQaYx8JqdjZMM0FJ1hDOxLia_RUwfw3T7HJYzZCk3NQvzTGOzKOAebQ0MlYXLMtJciPzCGAeCSOzE4wMc7in6XxkcEvMw4B5OCwAN4ZwRhHXe3Ou9h885BTgIcA8NJoCFBPGAyYrMEbp5TN_AcxfShhUZILi-mkW4x7AMsAsC8_MIbBv5jR3OL3Y7EHAPBhWswdDpllYwesxw4jN2o7MA4B5IOzOQKApi17W7Lrfa277zNwPmPvDM6NGMUUmM666f2baYu4DzH2AWWcXeGaYdY53v5q1jQD-DJg_h_OZRSE5yynFIrm0-V7A3FvC9B5vKmzROabtNLsHMPeUphm3hZC6PgIYxjRrm2n-CTB_CnsA2wAGSUtRfdQcY6b5R8D8ETAnXwkRgClMnnd_2NR2ZP4AmD-EQBM2zm1WGJLX75sNY2TaRs2_B8zvAbNwAWyca2IVr89nhhE1t51mvwPM78LIPAKrWZZb4-qX5hjzmd8C5rch0IR8RiOj_S5gaiPzG8D8JiwAsG9WOEozWv_MxLgA3A2Yu0OguR8kZ723prH175lhLABtI4BfA-bXEAHMeBRGRmXUue43NNo-M78CzK_CyNwE7xlpmcnrX5ox7gH8EjC_HA1nyJ0PAQYbjUh9HUCMKcBdgLmrhLFFQWzDyVmMmDsBcydgVkyA2MwwlJv6veYYMb8AzC9KGOuI7AUAyWHuAMwd5ZEpJFeqPtCMEfNzwPy8hGFMG0LqV7MY85nbAXN7iM02DodN2kpTH87EmM_8DDA_C--Zm2EPwOVYNsRmMW413QaY28I0WxciAG4l1_V7ADE-M7cC5tYSBhWGGt39S7PtNPspYH4akrPHYEeTOexE92eabReAnwDmJ-GZeWYUozTBmNZPsxjT5lsAc0vAbBXCGYYy6ZJLzn4MmB8HDAnl84hS1H1y1jaf-RFgfgT5zKwnITlThhmU3vbsDwHzwxLGCsklrV8AYsT8ADA_KGFclttc1qfNw8C0Xc1uBszNYXfmilAJKDFl3VfPth2Z7wPm-zAy0-GwCWWZ4qr74rm2Z5o3AeamEsa5AquGKyfDONNsG87cCJgbIZxZPjcsALnluvsUoC3me4D5XglT8N7_-YalOUbMDYC5ATDPbgnTTFhnGo4BY8R8FzDfLWEypYU16e0BXA-Y60sY5IyhtL5GM0bMdwDznRJGak5yXr-axYi5DjDXAeZWHe7PMCESXM2-DZhvlzDUKSJFeiNzLWCuLWGU1rmR6e0BfAsw3yphMmdpXnQ_Mm2Ts2sAc004OYMKDZ5jR0n9BboYi7S_CZhvhk3A6-AyUC9sFiy9C3TfAMw3wu7Mi7CjiXvBTEP5_DBSgLZR89cB83WImmfuG_IZbDLd_S2NtvnM1wDzNcBMvgoWAC0YzbqvamqLuRowV5cwhltKsvRuNn0VMF8NW03nQ2ymJM5s9wtA22fmK4D5SnlkJJUE1y8AMVbPfhkwXw5L810QAbiCKpte7cyXAPOlsDQXsABwwzPd_fZs23DmKsBcBeHMZQymWSE40ellmksBs7SEcRwbg9OrA7gSMFdCwek2e4dODZajhiONGDFfBMwXS5hebCYsTe_KyRWAuaKE4TkrDO--7URbzBcA84USRnHjjEgP83nAfB4wO54WdjSdonl6q9nnAPO5gDkTFgCrmUTpFQJdDpjLAyb0nSmoFDy9erPLAHNZCSMNz7MxeGm2jc0-C5jPhthsj7BvplhTQ5AYo-bPAOYzEDU7uNigCXfG1k-zGPsBfBown4aRWXBYuHKCncnql-YYq5ouBcylIWrGoRRYEJPVb5zHWAdwCWAugWl29EwINBmXeAy6m7SdZhcD5uKQNkORtmA5sia9ivNPAeZTpZFhmrqCp1ehcRFgLiphKMukk-lhPgmYTwJmymUQATDLdV6_cR4j5kLAXFjCcFGwwnbfE7BtOPMJwHwCwpktLoSXZo5yNAYtwdpiLgDMBSUMdZbpPL0D2vMBcz5gtoYmOlYylpn0Lmp_HDAfL2FyXlhl6vfNYsScB5jzShite4uzSu-A9mOA-Vh5muHCyIZAM0bMuYA5FxocPnDJKAYLmamGY8AYMR8FzEfLGNqbZw3VszFizgHMOSVMLmmGG4rnYsScDZizSxikJM_G4GJD2xTgLMCcFVKA0KoFUU4bes8OIwVoizkTMGeGtPm5cLURZUakdxfgI4D5SBiZq2FpJjSjDWeawzhsaov5MGA-HDAbwF6z6KXNpP4C3TAwbVOAJYBZEooaboBAkxZWNDQEifFu8xmAOSNsAl4cdmfyHKP6FCDGKyenA-b0MM2g8xxD2IgE-wGcBpjTSkUN1BGBGi4Dxbg7cypgTg39AC4O27NK46x-AYjxLsApgDmljMG4MdCM8S7AyYA5ObQF3zMszVrQBGOzxYBZXMJYonPFUjpsUif1flUnrYBvz4wv9lo_9DbLHecp5TJqkYcsqkK4KAp_pyEhyEIPWViFOKl4LlI6YVYLPGTBKhBFCFHd9zNrATnRQ07sQw66J9zIkDpvCPcjg5zgISdUITZXGW04VIoMcryHHF-FZFqZzCW1as33kPlViMwEEg0tjOM651cf8pAPBcjIuEXnhpbf2ClW_4zEVeevjvOQ4_qQk-ZDniJzgVD9eySuQlI1z0Pm9SGnLgNITrWj9fFWXHm9mushc_uQxSfC6XFmCMlSyhzVsR5ybB-ycCT0XpaY0JSuj6g5HjJnAFkejiYyhVVKZSNqtofMHqxaBoJGzZXW9d3j4qpMUrM8ZNZKyIRjZsEL0TrKG766GFctv5rpITMHIwIliSgnymbd1_G3GJEZHjKjPyKTScgQs94f7b43eYsQxf-91fSVIQpZsSx8mMAx03CBJ7JYy_-qplUh2gink7q_r6Z6yNQ-5GnYFpLIZVokFf1O8ZApVQgS0sqkusSrYzzkmCpE0N5wFCn17VOTPWRyFZIx4XKVVIZ4tIccvRJCj7sbYi2TSaRSKtJRzkNcFaIwFjlOavm1HmKrEG0yrFlKNWC93KkHMVVITlHOx2CDrkUYrz1ED2KtE2BEmBSW1WeIcZ3JK-Uhqg85-RRYfglXWKZUXKCkh8hBqhs6jAjhXEOqG1li9UEP-eBg8-HwEMY7ljW8RyJLdY_ykKMGqe7UcBVXU9fw8afIptYHPOQDg2fk9NC-DmWk4UvWkU2tIz3kyMGqdT8UrCkrdcO3xeK65qGO8JAj-hnidGgpzCRBnNeHKHEdVav3e8j7-5ApAgoIsGASd9-yrsWIvM9D3leFSG1ULyVJqOZevddD3rsyREEvLIRnhCiHeVJh_OEecngVInsrlk3rDPEwDzlsJSTfZGmAaJmJpDLEQz3k0CqEGEtzm1Ilt3qPh7ynCimURK7hHldkkEM85JBVphbDhDV02Y0McrCHHNyHbPg4vEdyhPKGzwZHdhh6kIcc1A9RFmwWPrCRy4zVP-xx9ddXB3rIgdURyZSmliZV5nSAhxxQhVjNmR6D-_Qtwvj9PWT_wdS6F1YtnKFCdP_1thbPyH4est8gjIdbdMwXPpikjt7e7SHvHhy9wTUNmWspXFJHb-_ykHcNRmRHgDhcSNd9q8YWz8i-HrJvFWKk0lx2n-q2qHx4p4e8c7AddBu8ELXEVHdfwtFiau3jIfsMVq0jYKfRZobr7juAtMgQ9_aQvQcP-8rvgGmCi-6_aNBiRN7hIe8YPCO7h95fwjrU_fLbYkTe7iFvH-TsF4RWWUgI1P0lmRYhyl4eslc_RNnhvnCrlDmbVEd2taeH7FmFWJZnNKk22eptHvK2PmT7pSGMxxKrpEZkDw_Zowox1umCJXUYuruH7F6FOJlZZpKqxN7NQ3arQrSjzmVJ5ey7esiufciBAFGKK1skVcKxi4fsUoX4L6-bIqmdxrd6yFurEIGl0yKp5fctHvKWKoT0Xoaq4Xg6spx9Zw_ZeRDGw0GPkkIUuvtO6y2Cxjd7yJsHkE1XVgchw-ufkcgyROEhYgDZFnZRCoTQGBQMtBgR7iF8cFthLzhnpzxDDRV0kY0I8xA2gFwLzwhT0uruP0jWAkI9hA4qH5bAe8QqaRpOdSObWoWHFH3IGeuFnUZeCJJSO0VFPIT0M8TjNwq7KM42bdBFds6OPQT3IdN2halFen-2oV93ZOfsuYfkVQiljBtRv_kQWc6OPAT1Q5Rdr4ecnWn_ye6UYq3MQ7IqhGeENLUaiQyyk4fs1Idsd39ommQpkkmlupM8ZFIVYhzz301ICTLRQyZWIRYpQ2RSqe6OHrJjFZKpnEua1MO-g4fsMDgMvTW0SaRENbTjjQyyvYdsX4VIiXPDk-r5sJ2HbFeFENf7N61alG09ZNtVRoQJZYvuC89ahPHbeMg2g_OREGtJhCjpvsVbi6Bxaw_ZulplyjWSuOj-HmKLw9CtPGSrQYa4Ava1dIF7K3Dnh6EtIFt6yJaDzQcoKTcYE9rQOyiuHnXqTR7ypv7UmjobXohGI4q677fdYkS28JAtBg_7HZCPEFkYUh-iRDYim3vI5tURMVrofAy-GtAC8kYPeWMVQnNVoIbD0Mggm3nIZoPj6QOgzIlYWzR8zDUyyKYesmkV4pTKFU-pH73axEM26ZeUL4FWCZrrXI7BsUILyBs85A0DyMxwWYwqIZMqzny9h7y-D1kKjcIUZUSl1brtdR7yuipEs1wUDbFWZJCNPWTjKkQ6i61LakRe6yGvrUJwLjR2Se2ibOQhG_UhT90QRoQbIpOCvMZDXlOFMOOKpstikUFe7SGvrkI0QblO6isA6lUe8qoqBBFHCtb9dlCLMH5DD9lwcMX1IggaJSNZw7cZIyuX3cBDNhikusfCm52jgo3B_ZEWI7K-h6w_SHWvgYc9d1Q0XN-LLGdfz0PWG0ytm2D5pa4gDd2cIssQJ3jIhH4YP-sW2PslijWVOUWWj7zSQ15ZhaBC4byhgi6yzYdXeMgrBg_7kZBYGVOIhgYvw5haLXYa1_WQdVcZkYxbUtQHjZEdT7_cQ14-gDwMQSO1mebdXwNvAXmZh7xsAHki9EXJERuDZ6RFiLKOh6zTD1GWTYKdRiE1S-vo7aUe8tIqhBPJlEvqoGdtD1m7D3l853CfnWiRVuXDSzzkJVUI4owhmlTBwFoeslYV4gpBecMGXWQQ_30FNb4KMZQK1tByJzLIOA8Z14c8zyAf6T3tsuHjqpFB1vSQNauQjGNO0mpvuIaHrLHKiEha8CKlVUuOqJGqwtisQGQIfRKm_A-WO5nR
panic: compression corrupt

goroutine 1 [running]:
main.main()
        /path/to/couchdb-prometheus-exporter/cli/update_seq/decode.go:17 +0xc5
exit status 2
```

**Comment by gesellix** [comment]
This might require a fix in the https://github.com/okeuday/erlang_go module. I'm going to file an issue or a pull request there.

**Comment by gesellix** [comment]
See https://github.com/okeuday/erlang_go/pull/3

Depending on how fast we're getting any feedback, I might use the vendored approach to release a fix for you. Please leave a note on how urgent this issue is for your use case.

**Comment by gesellix** [comment]
A fix is available with release v30.6.0. Please leave some feedback, if anything isn't working as expected.

**Comment by Bolebo** [comment]
Wow, thanks for the quick fix ! I really appreciate it.
I will try the fix today and I'll let you know.
